linode/manager
# Feat: [M3-6713] Plan availability notices for GPU

## Description 📝
This PR is a follow up to https://github.com/linode/manager/pull/9253 to add a dynamic version of the GPU Plans support. 

It also 
- fixes a regression with introduced in the previous PR where we would not get the default proper plan selected when cloning or using a backup
- fixes the styling (height) of the notice
- add the missing UI for the Linode Resize modal

## Preview 📷

| Before  | After   |
| ------- | ------- |
| ![Screenshot 2023-06-20 at 5 27 12 PM](https://github.com/linode/manager/assets/130582365/2dbe2783-b7b6-43de-b120-c0e5adcaeb4e) | ![Screenshot 2023-06-20 at 5 26 58 PM](https://github.com/linode/manager/assets/130582365/e77d62d9-0b7e-48a7-9007-2e33f1eea46d) |

## How to test 🧪

> ⚠️ **This PR affects ALL linode creation flows (including K8 and node pools). Each flow should be tested for consistency, on both desktop and mobile**

--------

The verification steps underneath should be performed on the following paths:
**Linode Create:**
- /linodes/create
- /linodes/create?type=One-Click
- /create?type=StackScripts
- /linodes/create?type=Images
- /linodes/create?type=Backups
- /create?type=Clone%20Linode
- //linodes/[id] click resize

**Kubernetes Create**
- /kubernetes/create

Existing Cluster Add node pool (you need to test this with both a cluster in a non-eligible region and a cluster in a **Premium Plans regions**
- /kubernetes/clusters/[id]/summary

### Verification steps
- Confirm this behavior on the GPU tab when no region is selected
- Confirm this behavior with a non eligible premium region (ex: Washington DC)
- Confirm no message is displayed and form is enabled with a premium plans region (ex: Newark)
- Confirm no regression to the existing premium plans tab 

Steps and UI may vary slightly depending on the creation flow (ex Backups) but the behavior should be consistent across all.
